src/analysis/explorerHomeView.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import {CredGraph} from "../core/credrank/credGraph";
+import {type TimestampMs} from "../util/timestamp";
+import {type Interval} from "../core/interval";
+import * as Grain from "../core/ledger/grain";
+
+export type DataOverTime = $ReadOnlyArray<number>;
+
+export type UserRow = {|
+  +name: string,
+  +credEarnedInTimeRange: number,
+  +grainHarvestedInTimeRange: Grain.Grain,
+  +grainBalance: Grain.Grain,
+  +allTimeCred: DataOverTime,
+  +category: UserCategory,
+|};
+
+export type UserCategory = "participant" | "bot" | "project" | "organization";
+
+export type ExplorerHomeView = {|
+  +credEarned: number,
+  +grainHarvested: Grain.Grain,
+  +participantCount: number, //Question: what is this supposed to capture? Is it supposed to change depending on the time range selected?
+  +grainPerCred: Grain.Grain,
+  +userRows: $ReadOnlyArray<UserRow>,
+  +intervalsIncluded: $ReadOnlyArray<Interval>,
+|};
+
+/**
+  Responsible for transforming data from the CredGraph and LedgerLog(?) into the information needed by the Explorer Home web page.
+ */
+export class ExplorerHomeViewBuilder {
+  constructor(credGraph: CredGraph) {
+    credGraph;
+    //Other params, depending on implementation. Probably a new GrainData API.
+    //Saves params or transforms them, depending on implementation strategy
+  }
+
+  /**
+    Returns all available intervals.
+    Useful for determining appropriate time ranges to offer for selection.
+   */
+  getAllIntervals(): $ReadOnlyArray<Interval> {
+    return [];
+  }
```

Comment:
this would actually return an `IntervalSequence` type. it's basically a ROA of intervals with some extra guarantees. 

https://github.com/sourcecred/sourcecred/blob/55c803ee60e1668d3d3262f0df096d6c30092c29/src/core/interval.js#L21-L29

Reply:
cool cool